integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/spi/DifferentSessionFactoriesIT.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.spi;
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.search.integrationtest.mapper.orm.automaticindexing.AutomaticIndexingBasicIT;
+import org.hibernate.search.mapper.orm.impl.HibernateSearchContextService;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmMapping;
+import org.hibernate.search.mapper.orm.mapping.HibernateOrmSearchManager;
+import org.hibernate.search.util.SearchException;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.orm.OrmSetupHelper;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Tests the creation of a {@link HibernateOrmSearchManager}, using a {@link HibernateOrmMapping} with an {@link EntityManager} owned by a different {@link SessionFactory}.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class DifferentSessionFactoriesIT {
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = new OrmSetupHelper();
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private SessionFactory sessionFactory;
+	private SessionFactory sessionFactoryAlt;
+	private EntityManager entityManager;
+
+	@Before
+	public void setup() {
+		sessionFactory = initABasicSessionFactory();
+		sessionFactoryAlt = initABasicSessionFactory();
+		entityManager = sessionFactory.createEntityManager();
+	}
+
+	@Test
+	public void tryToUseDifferentSessionFactories() {
+		thrown.expect( SearchException.class );
+		thrown.expectMessage( "Mapping service cannot create a search manager using a different session factory." );
+
+		// service mapping is taken from the alternative session factory
+		HibernateOrmMapping serviceMapping = sessionFactoryAlt.unwrap( SessionFactoryImplementor.class )
+			.getServiceRegistry().getService( HibernateSearchContextService.class ).getMapping();
+
+		// try to use an entityManager owned by the original session factory instead
+		serviceMapping.createSearchManager( entityManager );
+	}
+
+	@After
+	public void teardown() {
+		if ( entityManager != null ) {
+			entityManager.close();
+		}
+		if ( sessionFactory != null ) {
+			sessionFactory.close();
+		}
+		if ( sessionFactoryAlt != null ) {
+			sessionFactoryAlt.close();
+		}
+	}
+
+	private SessionFactory initABasicSessionFactory() {
+		backendMock.expectSchema( AutomaticIndexingBasicIT.IndexedEntity.INDEX, b -> b
```

Comment:
You can use `backendMock.expectAnySchema( IndexedEntity.INDEX )` here, since you don't really care about the content of the schema.

Reply:
Nice to weak the expectation here. Thanks